resolwe_bio/tools/expression_aggregator.py

```diff
@@ -0,0 +1,155 @@
+    parser.add_argument('-t', '--expression-type', help='Expression type', required=True)
+    parser.add_argument('-g', '--group-by', help='Group by', required=True)
+    parser.add_argument('-a', '--aggregator', help='Aggregator')
+    parser.add_argument('-b', '--output_box_plot', help='Output box plot')
+    parser.add_argument('-x', '--output_expressions', help='Output expressions')
+    return parser.parse_args()
+
+
+def load_expression(fn=None, sep='\t'):
+    """Read expressions from file."""
+    with gzip.open(fn, 'rt') as f:
+        reader = csv.DictReader(f, delimiter=sep)
+        return {row['Gene']: float(row['Expression']) for row in reader}
+
+
+def get_indices(descriptors, descriptor):
+    """Return positions of my_element in my_list."""
+    return {i for i, x in enumerate(descriptors) if x == descriptor}
+
+
+def get_values(expressions, descriptors, gene, descriptor):
+    """Return expressions of a gene with matching descriptor."""
+    indices = get_indices(descriptors, descriptor)
+    return [expression[gene] for i, expression in enumerate(expressions) if i in indices and gene in expression]
+
+
+def load_expressions(aggregator, expression_fns=[], sep='\t', descriptors=[]):
+    """Read expressions from files."""
+    raw_expressions = [load_expression(expression_fn, sep) for expression_fn in expression_fns]
+    if aggregator:
+        raw_expressions.extend(aggregator['raw_expressions'])
+        descriptors.extend(aggregator['descriptors'])
+    genes = {key for raw_expression in raw_expressions for key in raw_expression.keys()}
+    return raw_expressions, descriptors, {
+        gene: {
+            descriptor: get_values(raw_expressions, descriptors, gene, descriptor)
+            for descriptor in set(descriptors)
+            if get_values(raw_expressions, descriptors, gene, descriptor)
+        }
+        for gene in genes
+    }
+
+
+def generate_statistic(expression, gene, attribute, expression_type):
+    """Get box plot statistic for expressions of a single gene and attribute."""
+    min_val = min(expression)
+    max_val = max(expression)
+    median = np.percentile(expression, 50.0)
+    q1 = np.percentile(expression, 25.0)
+    q3 = np.percentile(expression, 75.0)
+    iqr = q3 - q1
+    lowerwhisker = max(min_val, q1 - 1.5 * iqr)
+    upperwhisker = min(max_val, q3 + 1.5 * iqr)
+    return {
+        'attribute': attribute,
+        'gene': gene,
+        'exp_types': [expression_type],
+        'min': min_val,
+        'max': max_val,
+        'median': median,
+        'q1': q1,
+        'q3': q3,
+        'lowerwhisker': lowerwhisker,
+        'upperwhisker': upperwhisker
+    }
+
+
+def get_statistics(expressions, expression_type):
+    """Get box plot statistics for expressions of all genes and attributes."""
+    return {
+        gene: [generate_statistic(exp, gene, descriptor, expression_type) for descriptor, exp in expression.items()]
```

Comment:
Split this into multiple lines:

    gene: [
        generate_statistic(exp, gene, descriptor, expression_type) 
        for descriptor, exp in expression.items()
    ]